frontend/src/modules/experimentRunLanes/services.ts

```diff
@@ -0,0 +1,81 @@
+import { FMSMetric } from "../../models/fms_api_models"
+import { Dataset } from "../../models/frontend_models"
+import store from "../../store"
+import api from "../../utils/api"
+import { DatasetInfo, ExperimentRunLanes, NumberOfReads, ReadsPerSample, ValidationStatus } from "./models"
+
+
+export async function loadExperimentRunLanes(experimentRunName: string) {
+    /*
+        Request all the datasets associated with the experiment run.
+        Group them by lane number.
+        Determine the list of lanes for the experiment.
+        For each lane:
+            Request the validation status for the lane
+    */
+    // Request all of the datasets associated with the experiment run
+    const datasetOptions = {
+        run_name: experimentRunName
+    }
+    const response = await store.dispatch(api.datasets.list(datasetOptions))
```

Comment:
why do you sometimes `useAppDispatch` and other times use the `store.dispatch` directly?

Reply:
useAppDispatch is a hook that can only be used from components. Services are just plain functions, so I have to dispatch actions using the imported `store`. I'm only dispatching actions to be able to use the api, which is based on thunks that have to be dispatched to be executed.